Source/WebKit/WebProcess/UserContent/WebUserContentController.cpp

```diff
@@ -417,7 +418,12 @@ void WebUserContentController::addUserScriptInternal(InjectedBundleScriptWorld&
 
             auto& mainFrame = page.mainFrame();
             if (userScript.injectedFrames() == UserContentInjectedFrames::InjectInTopFrameOnly) {
-                mainFrame.injectUserScriptImmediately(world.coreWorld(), userScript);
+                for (auto* frame = &mainFrame; frame; frame = frame->tree().traverseNext(&mainFrame)) {
```

Comment:
the `&mainFrame` parameter to `traverseNext()` is not needed since you are traversing the whole frame tree.